core/trino-main/src/main/java/io/trino/metadata/DiscoveryNodeManager.java

```diff
@@ -93,6 +97,10 @@
     @GuardedBy("this")
     private Set<InternalNode> coordinators;
 
+    // nodesToExclude contains nodes to be decommissioned.
+    @GuardedBy("this")
+    private Set<String> nodesToExclude = new HashSet<>();
```

Comment:
maybe name it nodesToBeDecomissined or nodesWithDecomissioningRequested? Longer but we keep the nomenclature simpler (we do not introduce "excluded" which is another "state"/"state description"

Reply:
done